### ISSUE TITLE
Maybe rename `TextUiWriter` to `UiTextWriter`

### ISSUE THREAD

**Issue by Olle-Lukowski** [bevyengine/bevy#16394]
In bevyengine/bevy-website#1782, it was mentioned that the (WIP) migration guide for the text rework (#15591), refers to
`UiTextWriter` instead of the actual name we have right now `TextUiWriter`.

I think it is obvious how this mistake slipped in there, `UiTextWriter` is a more intuitive name. It is (in my opinion) better to have a type alias' name have something appended or prepended to the name, instead of plopping something in the middle (like we have right now, where we added `Ui` in between instead of at the front).

The same thing goes for `TextUiReader`, I think they should both be renamed.

Let me know what you think, just wanted to ask this before I start writing some stuff for the website.

**Comment by rparrett** [comment]
These were renamed on purpose in https://github.com/bevyengine/bevy/pull/15887. Its migration is meant to guide section is meant to replace the one for #15591.

**Comment by ickshonpe** [comment]
Yeah I didn't like the new `Text<X>Reader/Writer` naming scheme from the PR but didn't feel like it was important enough to fuss over. 
Definitely strongly prefer prefixes or suffixes over infixes for name discriminators.

**Comment by UkoeHB** [comment]
Yep also not a fan of the change, but an executive decision was made. Just forgot to update the migration guide amidst the hustle to get that PR out.
